src/modules/NtpClient.rb

```diff
@@ -266,92 +253,59 @@ def GetCountryNames
     def GetNtpServersByCountry(country, terse_output)
       country_names = {}
       servers = GetNtpServers()
-      if country != ""
-        servers = Builtins.filter(servers) do |s, o|
-          Ops.get(o, "country", "") == country
-        end
+      if country.to_s != ""
+        servers.select! { |_server, attrs| attrs["country"] == country }
         # bnc#458917 add country, in case data/country.ycp does not have it
-        p = MakePoolRecord(country, "")
-        Ops.set(servers, Ops.get(p, "address", ""), p)
+        pool_country_record = MakePoolRecord(country, "")
+        servers[pool_country_record["address"]] = pool_country_record
       else
         country_names = GetCountryNames()
       end
 
-      default_already_chosen = false
-      items = Builtins.maplist(servers) do |s, o|
-        label = Ops.get(o, "location", "")
-        l_country = Ops.get(o, "country", "")
-        if country != ""
-          l_country = ""
-        else
-          l_country = Ops.get(country_names, l_country, l_country)
-        end
-        if label != "" && l_country != ""
-          label = Builtins.sformat("%1 (%2, %3)", s, label, l_country)
-        elsif label == "" && l_country == ""
-          label = s
-        else
-          label = Builtins.sformat("%1 (%2%3)", s, label, l_country)
-        end
-
+      default = false
+      items = servers.map do |server, attrs|
```

Comment:
If this `map` is the last call of the method, you can avoid using an `items` variable, right?